test/jdk/sun/security/provider/named/NamedKeys.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (c) 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 8347938 8347941
+ * @modules java.base/sun.security.pkcs
+ *          java.base/sun.security.x509
+ * @library /test/lib
+ * @summary check the Named***Key behavior
+ */
+import jdk.test.lib.Asserts;
+import jdk.test.lib.security.SeededSecureRandom;
+import sun.security.pkcs.NamedPKCS8Key;
+import sun.security.x509.NamedX509Key;
+
+import java.util.Arrays;
+
+public class NamedKeys {
+    public static void main(String[] args) throws Exception {
+
+        var r = SeededSecureRandom.one();
+        var raw = r.nBytes(32);
+
+        // Create a key using raw bytes
+        var sk = NamedPKCS8Key.internalCreate("SHA", "SHA-256", raw, null);
```

Comment:
Can you add a comment somewhere that the algorithms in this test don't mattter? I was wondering why you were using "SHA" here and I had to go look at the `NamedPKCS8Key` code to see that it wasn't testing the "SHA" algorithm.